### PR TITLE
[release-v1.6] [SRVKE-1400] Measure initialize offsets and schedule latency, move to use max fill up for scheduler

### PR DESCRIPTION
**Pull request by pierDipi**
Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

### CODE REVIEW

File: control-plane/pkg/reconciler/consumergroup/consumergroup.go [hunk 1]
```diff
@@ -60,8 +66,38 @@ import (
...
+	scheduleDistribution = view.Distribution(10, 100, 1000, 10000, 30000, 60000)
+
+	initializeOffsetsLatencyStat = stats.Int64("initialize_offsets_latency", "Latency of consumer group offsets initialization operations", stats.UnitMilliseconds)
+	// initializeOffsetsDistribution defines the bucket boundaries for the histogram of initialize offsets latency metric.
+	// Bucket boundaries are 10ms, 100ms, 1s, 10s, 30s and 60s.
+	initializeOffsetsDistribution = view.Distribution(10, 100, 1000, 10000, 30000, 60000)
 )
 
+func init() {
+	views := []*view.View{
```
Comment (matzew) [inline control-plane/pkg/reconciler/consumergroup/consumergroup.go:82]: we keep ths? or just now?

Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/consumergroup.go:82]: I would keep them

File: control-plane/pkg/apis/internals/kafka/eventing/v1alpha1/consumer_group_lifecycle.go [hunk 2]
```diff
@@ -61,7 +61,10 @@ func (cg *ConsumerGroup) MarkReconcileConsumersFailedCondition(condition *apis.C
 		condition.GetMessage(),
 	)
 
-	return fmt.Errorf("consumers aren't ready, %v: %v", condition.GetReason(), condition.GetMessage())
+	// It is "normal" to have non-ready consumers, and we will get notified when their status change,
+	// so we don't need to return an error here which causes the object to be queued with an
+	// exponentially increasing delay.
+	return nil
```
Comment (matzew) [inline control-plane/pkg/apis/internals/kafka/eventing/v1alpha1/consumer_group_lifecycle.go:67]: nice comment, and +1 on `nil`, and being more lenient

File: control-plane/pkg/reconciler/consumergroup/controller.go [hunk 3]
```diff
@@ -228,7 +228,7 @@ func createStatefulSetScheduler(ctx context.Context, c SchedulerConfig, lister s
 		lister,
 		c.RefreshPeriod,
 		c.Capacity,
-		"", //  scheduler.SchedulerPolicyType field only applicable for old scheduler policy
+		scheduler.MAXFILLUP, //  scheduler.SchedulerPolicyType field only applicable for old scheduler policy
```
Comment (matzew) [inline control-plane/pkg/reconciler/consumergroup/controller.go:231]: any implication here?

File: openshift/patches/fix_scheduler_and_autoscaler.patch [hunk 4]
```diff
@@ -0,0 +1,332 @@
+diff --git a/vendor/knative.dev/eventing/pkg/scheduler/state/helpers.go b/vendor/knative.dev/eventing/pkg/scheduler/state/helpers.go
+index 56ca82b8d..b37fc7997 100644
+--- a/vendor/knative.dev/eventing/pkg/scheduler/state/helpers.go
```
Comment (matzew) [inline openshift/patches/fix_scheduler_and_autoscaler.patch:3]: will this hit "upstream" (eventing core)? Or are we just on not "latest" version?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**